File: chat/src/main/java/org/thoughtcrime/securesms/mediasend/MediaSendPageFragment.java
package org.thoughtcrime.securesms.mediasend;

import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * A page that sits in the {@link}.
 */
public interface MediaSendPageFragment {

  @NonNull Uri getUri();

  void setUri(@NonNull Uri uri);

  @Nullable Object saveState();

  void restoreState(@NonNull Object state);

  void notifyHidden();
}

File: chat/src/main/java/com/difft/android/chat/mediasend/MediaSendPageFragment.kt
package com.difft.android.chat.mediasend

import android.net.Uri

/**
 * A page that sits in the media send flow.
 */
interface MediaSendPageFragment {

    fun getUri(): Uri

    fun setUri(uri: Uri)

    fun saveState(): Any?

    fun restoreState(state: Any)

    fun notifyHidden()
}
